Library/Homebrew/utils/linkage.rb

```diff
@@ -0,0 +1,16 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Utils
+  sig {
+    params(binary: T.any(String, Pathname), library: T.any(String, Pathname),
+           prefix: T.any(String, Pathname)).returns(T::Boolean)
+  }
+  def self.check_binary_linkage(binary, library, prefix = HOMEBREW_PREFIX)
```

Comment:
```suggestion
  def self.binary_linked_to_library?(binary, library, prefix = HOMEBREW_PREFIX)
```
given it returns a `T::Boolean`